resources/saas-boost-web.yaml

```diff
@@ -18,16 +18,143 @@ Parameters:
+                Action:
+                  - logs:PutLogEvents
+                Resource: !Sub arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:*:log-stream:*
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogStream
+                  - logs:DescribeLogStreams
+                Resource: !Sub arn:${AWS::Partition}:logs:${AWS::Region}:${AWS::AccountId}:log-group:*
+              - Effect: Allow
+                Action:
+                  - s3:listBucket
+                  - s3:GetBucketVersioning
+                Resource:
+                  - !Sub arn:${AWS::Partition}:s3:::${SaaSBoostBucket}
+                  - !Sub arn:${AWS::Partition}:s3:::${AdminWebBucket}
+              - Effect: Allow
+                Action:
+                  - s3:GetObject
+                  - s3:GetObjectVersion
+                Resource:
+                  - !Sub arn:${AWS::Partition}:s3:::${SaaSBoostBucket}/client/web/*
+              - Effect: Allow
+                Action:
+                  - s3:DeleteObject
+                  - s3:PutObject
+                  - s3:PutObjectAcl
+                  - s3:GetObject
+                  - s3:GetObjectVersion
+                Resource:
+                  - !Sub arn:${AWS::Partition}:s3:::${AdminWebBucket}/*
+  AdminWebCodeBuildLogs:
     Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/codebuild/sb-${Environment}-admin-web
+      RetentionInDays: 14
+  AdminWebCodeBuildProject:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Name: !Sub sb-${Environment}-admin-web
+      ServiceRole: !Ref AdminWebCodeBuildRole
+      TimeoutInMinutes: 10
+      Artifacts:
+        Type: NO_ARTIFACTS
+      Environment:
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/amazonlinux2-aarch64-standard:2.0
+        Type: ARM_CONTAINER
+        EnvironmentVariables:
+          - Name: SOURCE_BUCKET
+            Value: !Ref SaaSBoostBucket
+          - Name: WEBSITE_BUCKET
+            Value: !Ref AdminWebBucket
+          - Name: REACT_APP_AWS_REGION
+            Value: !Ref AWS::Region
+          - Name: REACT_APP_AWS_ACCOUNT
+            Value: !Ref AWS::AccountId
+          - Name: REACT_APP_ENVIRONMENT
+            Value: !Ref Environment
+          - Name: REACT_APP_SIGNOUT_URI
+            Value: !Ref AdminWebUrl
+          - Name: REACT_APP_CALLBACK_URI
+            Value: !Ref AdminWebUrl
+          - Name: REACT_APP_API_URI
+            Value: !Ref ApiGatewayUrl
+          - Name: REACT_APP_CLIENT_ID
+            Value: !Ref AdminWebClientId
+          - Name: REACT_APP_COGNITO_USERPOOL
+            Value: !Ref UserPoolId
+          - Name: REACT_APP_COGNITO_USERPOOL_BASE_URI
+            Value: !Ref UserPoolBaseUri
+      Source:
+        Type: NO_SOURCE
+        BuildSpec: |
+          version: 0.2
+          phases:
+            pre_build:
+              commands:
+                - n 14
```

Comment:
I assume this ensures that node 14 is installed, but I can't find any docs. Am I right in assuming this?

Reply:
Yes, it's the Node version management system. It will download and install the version requested if it doesn't already exist and will switch the Node binaries on the `$PATH` to the version of your choice.